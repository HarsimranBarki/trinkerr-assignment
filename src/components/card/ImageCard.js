import {
  motion,
  useAnimation,
  useMotionValue,
  useTransform,
} from "framer-motion";

import React from "react";

const ImageCard = ({ url, name, handleImage }) => {
  const x = useMotionValue(0);
  const xRange = [-200, -100, 100, 200];
  const opacityRange = [0, 1, 1, 0];
  const opacity = useTransform(x, xRange, opacityRange);
  // Framer animation hook
  const animControls = useAnimation();

  return (
    <motion.img
      src={url}
      center
      // Card can be drag only on x-axis
      data-testid="cardImage"
      style={{ opacity, x }}
      drag="x"
      name={name}
      alt={name}
      dragElastic={1}
      dragConstraints={{ left: 0, right: 0 }}
      onDragEnd={(event, info) => {
        // If the card is dragged only upto 150 on x-axis
        // bring it back to initial position
        if (Math.abs(info.point.x) <= 150) {
          animControls.start({ x: 0 });
          handleImage("left");
          console.log("Left", info.point.x);
        } else {
          // If card is dragged beyond 150
          // make it disappear

          // Making use of ternary operator
          console.log("Right", info.point.x);
          animControls.start({ x: info.point.x < 0 ? -200 : 200 });

          handleImage("right");
        }
      }}
      className="  max-h-sm w-100 sm:max-w-sm  bg-blue-100 p-5 mt-5 shadow-md swipe-image"
    />
  );
};

export default ImageCard;
