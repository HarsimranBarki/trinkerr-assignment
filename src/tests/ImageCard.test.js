import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import React from "react";
import ImageCard from "../components/card/ImageCard";

describe("ImageCard", () => {
  it("ImageCard: should render correctly", async () => {
    const tree = renderer.create(<ImageCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  const handleImage = jest.fn();
  it("ImageCard: should have correct passed props ( :img :url :name )", () => {
    render(
      <ImageCard
        handleImage={handleImage}
        url={"https://www.google.com/"}
        name={"Google Image"}
      />
    );

    const imageElement = screen.queryByRole("img");
    const imageSrc = imageElement.src;
    const imageName = imageElement.name;
    expect(imageSrc).toBe("https://www.google.com/");
    expect(imageName).toBe("Google Image");
  });
});
