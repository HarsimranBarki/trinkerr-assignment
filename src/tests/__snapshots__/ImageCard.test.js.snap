// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCard ImageCard: should render correctly 1`] = `
<img
  className="  max-h-sm w-100 sm:max-w-sm  bg-blue-100 p-5 mt-5 shadow-md swipe-image"
  data-testid="cardImage"
  draggable={false}
  style={
    Object {
      "WebkitTouchCallout": "none",
      "WebkitUserSelect": "none",
      "opacity": 1,
      "touchAction": "pan-y",
      "transform": "none",
      "userSelect": "none",
    }
  }
/>
`;
